packages/scxa-experiment-page/src/TSnePlotViewRoute.js

```diff
@@ -0,0 +1,338 @@
+
+import AnatomogramCellTypeHeatmapView from './results/AnatomogramCellTypeHeatmapView'
+
+import TSnePlotView from '@ebi-gene-expression-group/scxa-tsne-plot'
+import { ClustersHeatmapView }  from '@ebi-gene-expression-group/scxa-marker-gene-heatmap'
+
+import BioentityInformation from '@ebi-gene-expression-group/atlas-bioentity-information'
+import { withFetchLoader } from '@ebi-gene-expression-group/atlas-react-fetch-loader'
+
+import {intersection as _intersection, first as _first, map as _map} from 'lodash'
+
+const BioentityInformationWithFetchLoader = withFetchLoader(BioentityInformation)
+
+const RedirectWithSearchAndHash = (props) =>
+  <Redirect to={{ pathname: props.pathname, search: props.location.search, hash: props.location.hash}} />
+
+const RedirectWithLocation = withRouter(RedirectWithSearchAndHash)
+
+const CLUSTERS_PLOT = `clusters`
+const METADATA_PLOT = `metadata`
+const CELL_TYPE_MARKER_GENE_HEATMAP = `celltypes`
+const CLUSTER_MARKER_GENE_HEATMAP = `clusters`
+
+class TSnePlotViewRoute extends React.Component {
+  constructor(props) {
+      super(props)
+    const cellTypeValue = _first(_intersection(_map(this.props.metadata,`label`), this.props.initialCellTypeValues))
+    const search = URI(this.props.location.search).search(true)
+    this.state = {
+      selectedPlotType: Object.keys(this.props.defaultPlotMethodAndParameterisation)[0],
+      geneId: ``,
+      selectedPlotOption: Object.values(Object.values(this.props.defaultPlotMethodAndParameterisation)[0])[0],
+      selectedPlotOptionLabel: Object.keys(Object.values(this.props.defaultPlotMethodAndParameterisation)[0])[0] + ": " +
+          Object.values(Object.values(this.props.defaultPlotMethodAndParameterisation)[0])[0],
+      selectedColourBy: cellTypeValue ? cellTypeValue.toLowerCase() :
+          this.props.ks.length > 0 ? this.props.ks[Math.round((this.props.ks.length -1) / 2)].toString() : ``,
+      highlightClusters: [],
+      experimentAccession: this.props.experimentAccession,
+      selectedColourByCategory: isNaN(search.k) ? METADATA_PLOT :  CLUSTERS_PLOT,
+      selectedClusterId: cellTypeValue ? cellTypeValue.toLowerCase() :
+          this.props.ks.length > 0 ? this.props.ks[Math.round((this.props.ks.length -1) / 2)].toString() : ``,
+      selectedClusterIdOption: ``
+      }
+  }
+
+  render() {
+    const { location, match, history } = this.props
+    const { atlasUrl, suggesterEndpoint, defaultPlotMethodAndParameterisation } = this.props
+    const { species, experimentAccession, accessKey, ks, ksWithMarkerGenes, plotTypesAndOptions, metadata, anatomogram } = this.props
+    const search = URI(location.search).search(true)
+
+    let plotTypeDropdown =
+      Object.keys(defaultPlotMethodAndParameterisation)
+        .map(plot => ({
+          plotType: plot,
+          plotOptions: plotTypesAndOptions[plot]
+        }))
+
+    let organWithMostOntologies = Object.keys(anatomogram)[0]
```

Comment:
This code snippet will not be resued, so I don't think it is good to create a function just because we need to clarify what is happening. I will add some comments instead to help understand.